source/guides/basics/loops.md

```diff
@@ -1,29 +1,88 @@
 Do That Again: Loops (PARTIAL)
 --------------------
 
-**TODO: Write-up that introduces loops in general, using the repeat command.**
+### What Is A 'Loop'?
 
-**TODO: Write-up explaining the foreach command.**
+All the scripts up until now had a single directional flow from top to bottom. An exact order was listed and were executed one after the other, with maybe skipping a few if you had an `if` command. However, what if you wanted to do that script you just wrote again? Well you have a few options. The most straightforward approach would simply be writing that same bit of script again. While this may work if you only wanted it repeated twice, or if there is only one line you want repeated, this can get out of hand really quickly.
 
-**TODO: Write-up explaining the while command.**
+Luckily you don't have to do this, Denizen has 3 different loop commands that repeat a block of script:
+- `repeat` to repeat a block a set amount of times
+- `foreach` to repeat a block based on a list input
+- `while` keep repeating a block until a condition is met
 
-### Placeholder
+### The Repeat Command
 
-Until this page is written, you can view the [old tutorial video here](https://one.denizenscript.com/denizen/vids/Loops).
+The most basic use of a loop is to simply just do a block of script again, in which the repeat command is best suited for.
+The basic format of a repeat command is
+```
+- repeat (number of times):
+    - (commands here)
+    - (will loop number of times)
+```
+
+This structure is similar to the `if` command, where you end the `repeat` command with a `:`. The commands that get looped need to also be indented in to let denizen know that these are the commands you want to be looped.
+
+Here is what it would look like in a real script...
+```dscript_green
+my_zap_task:
+    type: task
+    script:
+    - repeat 5:
+        - strike <player.location>
+        - narrate "you have been zapped!"
+        - wait 1s
+    - narrate "no more zaps"
```

Comment:
`zap` is a command in Denizen, a different word should be used here to avoid risk of confusion - probably `lightning` / `you have been struck by lightning`